cxx-checks/src/main/java/org/sonar/cxx/checks/HardcodedAccountCheck.java

```diff
@@ -55,45 +57,35 @@
    *
    */
   private static final String DEFAULT_REGULAR_EXPRESSION = "\\bDSN\\b.*=.*;\\b(UID|PWD)\\b=.*;";
-  private static Matcher reg;
 
-  /**
-   * regularExpression
-   *
-   */
   @RuleProperty(
     key = "regularExpression",
     description = "literal regular expression rule",
     defaultValue = DEFAULT_REGULAR_EXPRESSION)
-  public String regularExpression = DEFAULT_REGULAR_EXPRESSION;
+  private String regularExpression = DEFAULT_REGULAR_EXPRESSION;
 
-  public String getRegularExpression() {
-    return regularExpression;
-  }
+  private Pattern pattern;
 
-  @SuppressWarnings("squid:S2696") // ... initialize SquidAstVisitor
   @Override
   public void init() {
-    String regEx = getRegularExpression();
     Objects.requireNonNull(regularExpression, "getRegularExpression() should not return null");
 
-    if (null != regEx && !regEx.isEmpty()) {
+    if (!regularExpression.isEmpty()) {
```

Comment:
probably throwing an `IllegalStateException` in case of empty regular expression would be more correct

Reply:
@ivangalkin general rule we had in the past:
* configuration errors should result in errors and "stop" the analysis
* runtime errors: https://github.com/SonarOpenCommunity/sonar-cxx/wiki/Supported-configuration-properties#sonarcxxerrorrecoveryenabled

Wrong regex is configuration error.

Reply:
Ok. Then I'll throw an exception if regular expression was empty